ci/README.md

```diff
@@ -106,3 +106,97 @@ Therefore, the vast majority of releases are made using a target commit from a
 
 Release branches do not run CI on their own commits - instead, CI is run on PRs
 targeting them, and we enforce linear merges.
+
+## Working with Azure Pipelines
+
+### Understanding what gets built
+
+Azure Pipelines does not build your branch or your PR; instead, what it builds
+is the result of merging your branch into its target (in most cases, `main` or
+`main-2.x`). This has some nice properties (you don't need to explicitly
+rebase/merge to be confident your PR builds against current head), but it can
+also cause some subtle issues because **this is done per job**.
+
+Meaning that, within a single build, two separate jobs may not be building the
+same code. This is particularly problematic for the platform-independence test,
+in rare cases where the various platform jobs don't start at the same time and
+the changes on main in-between change the produced DAR file.
+
+### Restarting a failed build
+
+Azure Pipelines should trigger a build on every pull request (but not every
+branch). If a build has failed and you believe the failure to be flaky, you can
+re-run the build by navigating to the "Checks" tab of your PR, and clicking the
+"Re-run failed checks" button in the top right.
```

Comment:
Maybe also worthwhile to mention the `/azp run` commenting on a PR to re-run the build?

Reply:
Oh right, thanks! I'll add that.